nlp/2021_11_25_gpt3mix/nlp_gpt3mix.ipynb

```diff
@@ -0,0 +1,1966 @@
+      ],
+      "execution_count": null,
+      "outputs": []
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "Ir6zD__FU84n"
+      },
+      "source": [
+        "### Imports"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "metadata": {
+        "id": "hM-09Chsj7wI"
+      },
+      "source": [
+        "import re\n",
+        "import json\n",
+        "import torch\n",
+        "import random\n",
+        "import requests\n",
+        "import numpy as np\n",
+        "import pandas as pd\n",
+        "\n",
+        "import plotly.express as px\n",
+        "import plotly.graph_objects as go\n",
+        "from plotly.subplots import make_subplots\n",
+        "\n",
+        "from datasets import load_dataset, concatenate_datasets, load_from_disk, load_metric, Dataset\n",
+        "from transformers import pipeline, AutoTokenizer, AutoModelForSequenceClassification, TrainingArguments, Trainer, TrainerCallback, AutoModelForCausalLM, AutoModelForSeq2SeqLM"
+      ],
+      "execution_count": null,
+      "outputs": []
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "gFQ6LEgjeRWb"
+      },
+      "source": [
+        "### Download dataset\n",
+        "We'll train and evaluate our models on [Emotion](https://huggingface.co/datasets/emotion) dataset that contains English Twitter messages labeled as one of the six basic emotions: anger, fear, joy, love, sadness and surprise. To reduce the complexity of the task, we will keep only three labels, namely:\n",
+        "- joy 😂\n",
+        "- anger 😠\n",
+        "- surprise 😯"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "metadata": {
+        "id": "1ltsurmvaaGd"
+      },
+      "source": [
+        "# load the dataset and filter on samples that have a token count less than 30 to use only short tweets\n",
+        "max_input_len = 30\n",
+        "tokenizer = AutoTokenizer.from_pretrained(\"distilbert-base-uncased\")\n",
+        "emotion_ds = load_dataset(\"emotion\").filter(lambda e: len(tokenizer.batch_encode_plus([e['text']]).input_ids[0]) < int(max_input_len))"
+      ],
+      "execution_count": null,
+      "outputs": []
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "RQ24-mWS6elF"
+      },
+      "source": [
+        "The dataset is already split into 16,000 train and 2,000 test samples. To investigate the effectiveness of the augmentation method, we will use only 10 samples per class as a train set."
+      ]
+    },
+    {
+      "cell_type": "code",
+      "metadata": {
+        "id": "9YI6-pZdSqnI"
+      },
+      "source": [
+        "# select 10 random train samples from each of the three emotions\n",
+        "# sadness (0), joy (1), love (2), anger (3), fear (4), surprise (5)\n",
+        "joy_train_samples = emotion_ds['train'].filter(lambda e: e['label'] == 1).select(range(10))\n",
```

Comment:
Nitpick: but just selecting without shuffling won't select 10 at random, but just select the 10 first ones.